pkg/devserver/api.go

```diff
@@ -173,6 +174,13 @@ func (a devapi) register(ctx context.Context, r sdk.RegisterRequest) (err error)
 		Checksum: sum,
 	}
 
+	// Initially, we must delete all functions because we're straight up replacing them.
+	// This allows us to clean functions that are removed. Functions have a deterministic ID
+	// and so logs etc. are all still persisted.
+	if err := a.devserver.data.DeleteFunctionsByAppID(ctx, appParams.ID); err != nil {
```

Comment:
shouldn't this be inside the transaction as well?

Reply:
DuckDB doesn't handle txns with primary keys, but I've fixed this by loosening constraints.

Reply:
this is not the kind of data that needs to be in a columnal DB. I wonder if cqrs should have separate interfaces for relational DB vs columnal DB.